packages/shared/invokeGuardedCallbackImpl.js

```diff
@@ -53,10 +60,11 @@ if (__DEV__) {
   // Check that the browser supports the APIs we need to implement our special
   // DEV version of invokeGuardedCallback
   if (
-    typeof window !== 'undefined' &&
-    typeof window.dispatchEvent === 'function' &&
-    typeof document !== 'undefined' &&
-    typeof document.createEvent === 'function'
+    dispatchEvent &&
```

Comment:
We're not super consistent about this, but we try to use explicit `===` checks when possible to avoid accidents caused by type coercion. When it's a value/type we control, we tend to prefer `T | null` types and `=== null` checks, since that allows Flow to distinguish between intentionally missing and accidentally missing values.

Reply:
> Can we move the global module into `react-dom`, `scheduler`, etc as appropriate?

Makes sense. Though some of the files breaking this rulke would include `packages/shared/invokeGuardedCallbackImpl.js` inside of the `shared` package which is used by `ReactErrorUtils` which is used in multiple packages. Would the long term strategy be to have a copy of ReactErrorUtils in each package too?